### ISSUE TITLE
Git-Mediawiki is looking for a new maintainer

### ISSUE THREAD

**Issue by moy** [moy/Git-Mediawiki#33]
I'm not finding enough time to take good care of Git-Mediawiki these days, and I'm not using it a lot myself. It would be nice if someone would volunteer to take over the maintenance (alone, or together with me).

Let me know if you're interested.

**Comment by azbarcea** [comment]
I just found this tool this week, and I already submitted a patch on the git mailing list.

If you are still interested, I volunteer.

Btw, thank you for this tool.

**Comment by moy** [comment]
OK, for a start, I moved the repository from my account to a new organization, and invited you to the organization. I'm still there, but that should give you the permission needed to revive the project.

Thanks!

**Comment by anarcat** [comment]
i'm interested too, let's revive this project!

**Comment by moy** [comment]
I just sent you an invitation.

**Comment by azbarcea** [comment]
@moy, please send me an invitation too.

**Comment by moy** [comment]
You had an invitation pending for a while, I've just sent you another one.

**Comment by anarcat** [comment]
i've been accepted as a maintainer, thanks @moy!

**Comment by moy** [comment]
Just to clarify: @anarcat accepted to be an interim maintainer (and is doing a great job at reviving the project), but probably won't have much more time than me to spend on the project in the long run. A more active maintainer is still needed.

@anarcat: feel free to correct me if I misunderstood.

**Comment by anarcat** [comment]
On 2017-11-07 10:55:34, Matthieu Moy wrote:
> @anarcat: feel free to correct me if I misunderstood.

no, that's correct!

**Comment by hexmode** [comment]
I've forked this project and made several improvements before I was aware of anything going on here: https://github.com/hexmode/mediawiki-git-remote

**Comment by anarcat** [comment]
On 2017-11-20 17:48:37, Mark A. Hershberger wrote:
> I've forked this project and made several improvements before I was aware of anything going on here: https://github.com/hexmode/mediawiki-git-remote 

interesting!

how should we proceed to merge this back in? could you craft a series of
pull requests to converge here?

**Comment by hexmode** [comment]
anarcat <notifications@github.com> writes:

> how should we proceed to merge this back in? could you craft a series of
> pull requests to converge here?

First things first: could you try it out and see if it is any good from
your pov?

I want both of us to have motivation to get this in before I spend too
much energy on it.

**Comment by anarcat** [comment]
On 2017-11-20 19:34:31, Mark A. Hershberger wrote:
> anarcat <notifications@github.com> writes:
>
>> how should we proceed to merge this back in? could you craft a series of
>> pull requests to converge here?
>
> First things first: could you try it out and see if it is any good from
> your pov?
>
> I want both of us to have motivation to get this in before I spend too
> much energy on it.

At first glance, you'd need to work on the commitlogs in your pull
requests.

I'm not sure I have time to work on actually testing the code right now,
unfortunately. But I will definitely review any pull request you send.

**Comment by hexmode** [comment]
I appreciate that.  I wasn't creating code that other people would review. Still, getting your input on the how the code works would be good.

I'm slammed with other work ATM, so I'm not sure when I'll be able to get you the pull requests.

**Comment by anarcat** [comment]
On 2017-11-20 20:02:47, Mark A. Hershberger wrote:
> I appreciate that.  I wasn't creating code that other people would review. Still, getting your input on the how the code works would be good.
>
> I'm slammed with other work ATM, so I'm not sure when I'll be able to get you the pull requests.

Anytime is good. As long as we don't diverge too much and it becomes
impossible for you to join back the collective effort, I think it's fine
to take your time. :)

**Comment by hexmode** [comment]
So, I and some others at @MWStake are very interested in this.  I'm doing a presentation at the end of this month at the [Wikimedia Hackathon](https://www.mediawiki.org/wiki/Wikimedia_Hackathon_2019) that will use this.  In the meantime, I'll clean up my commits on #67.

**Comment by xloem** [comment]
Hi, I am available to be a minor maintainer.  I propose we have more maintainers, not fewer, in the spirit of a wiki.  I would like to reach out to PR contributors to ask if they are available to dedicate a little time to maintaining the repo as a whole.  We could make an issue page of maintenance tasks, for example.

**Comment by cooljeanius** [comment]
so... was one ever found?

**Comment by hexmode** [comment]
> so... was one ever found?

one what?

I'm still interested in this project and like @xloem's idea.  I think a good place to start would be to implement some CI.

**Comment by cooljeanius** [comment]
> > so... was one ever found?
> 
> one what?

a maintainer; someone I can ask for help with issues I run into when trying to use this to export a mediawiki page as a git repository

**Comment by xloem** [comment]
I don't have the capacity I hoped I would have when I posted that, but the things I mentioned are things anybody can do to help this situation.

**Comment by moy** [comment]
@anarcat is the person closest to being the "maintainer" for Git-Mediawiki (thanks again for the work you managed to do!), but apparently didn't manage to get more bandwidth than me to allocate to the project :-\. So, no, we don't really have a maintainer for now.

**Comment by xloem** [comment]
are you available to add hexmode and me?  what do you think of the idea of collaborative maintenance, more like a wiki?

**Comment by moy** [comment]
I've added both of you as members for now. I'll give you "owner" permission if everything goes right (or if there's something you want to do that's not possible as member, let me know).

I'm all for collaborative work. But it's usually nice to have a small set of persons that "feel responsible" for the project (as opposed to occasional contributors).

**Comment by anarcat** [comment]
On 2022-01-03 06:29:33, Matthieu Moy wrote:
> I've added both of you as members for now. I'll give you "owner" permission if everything goes right (or if there's something you want to do that's not possible as member, let me know).
>
> I'm all for collaborative work. But it's usually nice to have a small set of persons that "feel responsible" for the project (as opposed to occasional contributors).

I'm not sure where we are in this thread anymore, but one key issue with
maintaining this code is that it lives both here and inside the upstream
core git repo, which makes maintenance particularly tricky.

I wonder if it might be worth splitting that part out so that we at
least have a clear signal upstream where this code is maintained (or
not).

And, unfortunately, still not much more cycles available here, although
if one of my core wikis switch to mediawiki (rumours! I deny
everything!) I might definitely get more involved again.

**Comment by moy** [comment]
I think it's time to remove Git-Mediawiki from the source tree of git.git. I liked the idea of giving the Git community an easy access to the tool, but in practice this did not bring any real benefit, and puts more load on the maintainer. If we consider this repo as the official source of truth, and forget about git.git, it may help attracting a new contributor.

**Comment by anarcat** [comment]
On 2022-01-05 11:18:56, Matthieu Moy wrote:
> I think it's time to remove Git-Mediawiki from the source tree of git.git. I liked the idea of giving the Git community an easy access to the tool, but in practice this did not bring any real benefit, and puts more load on the maintainer. If we consider this repo as the official source of truth, and forget about git.git, it may help attracting a new contributor.

Sounds like a RFC PATCH to ***@***.*** Who's up for it? new issue maybe?

-- 
We reject kings, presidents and voting.
We believe in rough consensus and running code.
                        - David Clark

**Comment by lumeng** [comment]
> I've added both of you as members for now. I'll give you "owner" permission if everything goes right (or if there's something you want to do that's not possible as member, let me know).
> 
> I'm all for collaborative work. But it's usually nice to have a small set of persons that "feel responsible" for the project (as opposed to occasional contributors).

I am using this plug-in for MediaWiki instances as well. I'm interested to know more about its innerworkings and potentially help to maintain it. Is there an IRC channel (or other platform) for the developers and users of this software?

**Comment by moy** [comment]
> Is there an IRC channel (or other platform) for the developers and users of this software?

No, we're just using the issue tracker for communication. It's not like the community is that large ;-).

**Comment by hexmode** [comment]
My first pull request this time around: #74 

Whee!

**Comment by moy** [comment]
@hexmode has been active on the repo these days. I just gave owner permission on the organization.

**Comment by anarcat** [comment]
i wonder, should we close this issue then? maybe we're going to get more maintainers if we actually signal on the git upstream ML that we moved here? i filed #82 to discuss the formal retirement of mediawiki from upstream git...

**Comment by cooljeanius** [comment]
> i wonder, should we close this issue then?

I think that would make sense, yeah

**Comment by eli-schwartz** [comment]
> I think it's time to remove Git-Mediawiki from the source tree of git.git. I liked the idea of giving the Git community an easy access to the tool, but in practice this did not bring any real benefit, and puts more load on the maintainer. If we consider this repo as the official source of truth, and forget about git.git, it may help attracting a new contributor.

https://lore.kernel.org/git/20250512-pks-contrib-spring-cleanup-v3-6-32e151b0bfb0@pks.im/

As of git 2.51 it has been removed.
